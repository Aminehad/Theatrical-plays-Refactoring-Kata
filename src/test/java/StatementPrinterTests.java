import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;

import static org.approvaltests.Approvals.verify;
import static org.approvaltests.Approvals.verifyHtml;


public class StatementPrinterTests {

    @Test
    void exampleStatementPrinttotext() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", Play.playType.TRAGEDY));
        plays.put("as-like",  new Play("As You Like It",Play.playType.COMEDY));
        plays.put("othello",  new Play("Othello", Play.playType.TRAGEDY));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.print(invoice, plays);
        verify(result);
    }
      @Test
    void exampleStatementtohtml() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Play("Hamlet", Play.playType.TRAGEDY));
        plays.put("as-like",  new Play("As You Like It",Play.playType.COMEDY));
        plays.put("othello",  new Play("Othello", Play.playType.TRAGEDY));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.toHTML(invoice, plays);

        verifyHtml(result);
    }


}
