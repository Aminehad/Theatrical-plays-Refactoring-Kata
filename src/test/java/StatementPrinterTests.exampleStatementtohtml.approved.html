  Hamlet: $635.00 (55 seats)
  As You Like It: $580.00 (35 seats)
  Othello: $500.00 (40 seats)
<table>
<tr><th>Play</th><th>Seats</th><th>Amount</th></tr>
  <tr><td>Hamlet</td><td>55</td><td>$650.00</td></tr>
  <tr><td>As You Like It</td><td>35</td><td>$580.00</td></tr>
  <tr><td>Othello</td><td>40</td><td>$500.00</td></tr>
</table>
<p>Amount owed is $1,715.00</p>
<p>You earned 47 credits</p>
<p>Your points left 48 credits</p>
</body></html>
